vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.h

```diff
@@ -356,6 +360,8 @@ class LedgerImpl : public Ledger {
   uint32_t last_shown_tab_id_ = -1;
   std::queue<std::function<void()>> ready_callbacks_;
   ReadyState ready_state_ = ReadyState::kUninitialized;
+
+  FRIEND_TEST_ALL_PREFIXES(uphold::DisconnectWallet, Paths);
```

Comment:
Is this still necessary? If so, what private methods does that test need?

Reply:
Yep, `LedgerImpl::ready_state_` and `LedgerImpl::ReadyState`. 🙂

Reply:
Ah. We probably shouldn't have tests digging around in there. Instead of friending test classes, another approach frequently in Chromium is to have a (public) method that sets something up specifically for the test environment. Would something like this work?

```cpp
// public
void LedgerImpl::SetInitializedForTesting() {
  ready_state_ = ReadyState::kReady;
}
```

Longer term, we need to think about how we can structure `LedgerImpl` (and the codebase in general) to be more testable, and less reliant on pervasive mocking.

Reply:
How do you feel about
```
class LedgerImpl : public Ledger {
// ...
 protected:
  enum class ReadyState {
    kUninitialized,
    kInitializing,
    kReady,
    kShuttingDown
  } ready_state_ = ReadyState::kUninitialized;

 private:
// ...
};
```
and
```
class MockLedgerImpl : public LedgerImpl {
 public:
// ...
  void SetStateForTesting();
};
```

```
void MockLedgerImpl::SetStateForTesting() {
  ready_state_ = ReadyState::kReady;
}
```
? 🙂

Reply:
Personally, I'd probably rather put the `SetInitializedForTesting` on `LedgerImpl` and keep the state implementation private (since that makes it really clear why we need to expose it, and it leaks only the minimum required implementation detail). But I prefer either approach to test friendship, in this case. Your call!